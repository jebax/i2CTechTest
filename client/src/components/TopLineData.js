import React, { Component } from 'react'
import axios from 'axios'
import TopLineChart from './TopLineChart'
import Paper from '@material-ui/core/Paper'

export default class TopLineData extends Component {
  state = {
    spendData: [],
    unitsData: [],
    visitsData: [],
    totalCustsData: []
  }

  componentDidMount() {
    axios.get(process.env.REACT_APP_TOP_URL).then(response => {
      this.formatData(response.data)
    })
  }

  formatData = data => {
    let spendData = []
    let unitsData = []
    let visitsData = []
    let totalCustsData = []

    data.forEach(entry => {
      if (entry.METRIC === 'Spend') {
        this.appendFormattedData(spendData, entry)
      } else if (entry.METRIC === 'Units') {
        this.appendFormattedData(unitsData, entry)
      } else if (entry.METRIC === 'Visits') {
        this.appendFormattedData(visitsData, entry)
      } else {
        this.appendFormattedData(totalCustsData, entry)
      }
    })

    this.setState(prevState => ({
      spendData: spendData,
      unitsData: unitsData,
      visitsData: visitsData,
      totalCustsData: totalCustsData
    }))
  }

  appendFormattedData = (dataCollection, entry) => {
    dataCollection.push({
      x: entry.PRODUCT,
      y: entry.PCT_UPLIFT
    })
  }

  render() {
    return(
      <Paper elevation={3} id='topLineChart'>
        <TopLineChart
          spendData={this.state.spendData}
          unitsData={this.state.unitsData}
          visitsData={this.state.visitsData}
          totalCustsData={this.state.totalCustsData}
        />
      </Paper>
    )
  }
}
