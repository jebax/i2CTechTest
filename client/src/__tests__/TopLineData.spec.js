import React from 'react'
import { shallow } from 'enzyme'
import axios from 'axios'
import TopLineData from '../components/TopLineData'

jest.mock('axios')

describe('Top line data', () => {
  var topLineData
  var fetchedData

  beforeAll(() => {
    fetchedData = {

    }
    axios.get.mockResolvedValue(fetchedData)
    
    topLineData = shallow(<TopLineData />)
  })

  it('renders correctly', () => {
    expect(topLineData).toMatchSnapshot()
  })

  it('makes the correct API request', () => {
    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(axios.get).toHaveBeenCalledWith(process.env.REACT_APP_TOP_URL)
  })
})
