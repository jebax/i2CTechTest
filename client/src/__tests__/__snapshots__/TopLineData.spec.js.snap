// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top line data renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopLineData />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TopLineChart
        spendData={
          Array [
            Object {
              "x": "Aisle",
              "y": "0.01",
            },
          ]
        }
        totalCustsData={
          Array [
            Object {
              "x": "Aisle",
              "y": "0.04",
            },
          ]
        }
        unitsData={
          Array [
            Object {
              "x": "Aisle",
              "y": "0.02",
            },
          ]
        }
        visitsData={
          Array [
            Object {
              "x": "Aisle",
              "y": "0.03",
            },
          ]
        }
      />,
      "elevation": 3,
      "id": "topLineChart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "spendData": Array [
          Object {
            "x": "Aisle",
            "y": "0.01",
          },
        ],
        "totalCustsData": Array [
          Object {
            "x": "Aisle",
            "y": "0.04",
          },
        ],
        "unitsData": Array [
          Object {
            "x": "Aisle",
            "y": "0.02",
          },
        ],
        "visitsData": Array [
          Object {
            "x": "Aisle",
            "y": "0.03",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TopLineChart
          spendData={
            Array [
              Object {
                "x": "Aisle",
                "y": "0.01",
              },
            ]
          }
          totalCustsData={
            Array [
              Object {
                "x": "Aisle",
                "y": "0.04",
              },
            ]
          }
          unitsData={
            Array [
              Object {
                "x": "Aisle",
                "y": "0.02",
              },
            ]
          }
          visitsData={
            Array [
              Object {
                "x": "Aisle",
                "y": "0.03",
              },
            ]
          }
        />,
        "elevation": 3,
        "id": "topLineChart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "spendData": Array [
            Object {
              "x": "Aisle",
              "y": "0.01",
            },
          ],
          "totalCustsData": Array [
            Object {
              "x": "Aisle",
              "y": "0.04",
            },
          ],
          "unitsData": Array [
            Object {
              "x": "Aisle",
              "y": "0.02",
            },
          ],
          "visitsData": Array [
            Object {
              "x": "Aisle",
              "y": "0.03",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
